test/views/declaration/WarehouseDetailsViewSpec.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.declaration
+
+import base.Injector
+import forms.declaration.WarehouseDetails
+import models.Mode
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.test.Helpers.stubMessages
+import services.cache.ExportsTestData
+import unit.tools.Stubs
+import views.declaration.spec.UnitViewSpec
+import views.html.declaration.warehouse_details
+import views.tags.ViewTest
+
+@ViewTest
+class WarehouseDetailsViewSpec extends UnitViewSpec with ExportsTestData with Stubs with Injector {
+
+  private val page = new warehouse_details(mainTemplate)
+  private val form: Form[WarehouseDetails] = WarehouseDetails.form()
+
+  private def createView(mode: Mode = Mode.Normal, form: Form[WarehouseDetails] = form, messages: Messages = stubMessages()): Document =
+    page(mode, form)(journeyRequest(), messages)
+
+  "Warehouse Details View" should {
+    val view = createView()
+
+    "have proper messages for labels" in {
+      val messages = instanceOf[MessagesApi].preferred(journeyRequest())
+      messages must haveTranslationFor("supplementary.warehouse.title")
+    }
+
+    "display same page title as header" in {
+      val viewWithMessage = createView(messages = realMessagesApi.preferred(request))
+      viewWithMessage.title() must include(viewWithMessage.getElementsByTag("h1").text())
+    }
+
+    "display 'Back' button that links to 'Warehouse Identification Number' page" in {
+      val backButton = view.getElementById("link-back")
+
+      backButton.text() mustBe "site.back"
+      backButton.getElementById("link-back") must haveHref(controllers.declaration.routes.WarehouseIdentificationController.displayPage(Mode.Normal))
+    }
+
```

Comment:
Please add test for having inputs

Reply:
This page was simply renamed. It should be removed by the end of the sprint, so I am going to avoid spending time here.